### ISSUE TITLE
[Firestore] List collection ids of a document

### ISSUE THREAD

**Issue by guanzo** [googleapis/google-cloud-python#5480]
How do I list subcollection ids of a document? Is it not yet implemented in the python firestore library? It's available for all other google cloud platform languages: [node](https://cloud.google.com/nodejs/docs/reference/firestore/0.14.x/DocumentReference#getCollections), [go](https://godoc.org/cloud.google.com/go/firestore#DocumentRef.Collections), [php](http://googlecloudplatform.github.io/google-cloud-php/#/docs/cloud-firestore/v0.11.0/firestore/documentreference?method=collections), [java](https://googlecloudplatform.github.io/google-cloud-java/google-cloud-clients/apidocs/index.html?com/google/cloud/firestore/package-summary.html), [ruby](http://googlecloudplatform.github.io/google-cloud-ruby/#/docs/google-cloud-firestore/v0.21.1/google/cloud/firestore/documentreference?method=cols-instance), [.NET](http://googlecloudplatform.github.io/google-cloud-dotnet/docs/Google.Cloud.Firestore/api/Google.Cloud.Firestore.DocumentReference.html#Google_Cloud_Firestore_DocumentReference_ListCollectionsAsync). There's no way that the python library is the only library that doesn't support this functionality, right?

**Comment by tseaver** [comment]
@guanzo You are correct that the `DocumentReference` class does not expose a method to return subcollections.

**Comment by tseaver** [comment]
@jba the [docs for listing subcollections](https://cloud.google.com/firestore/docs/query-data/get-data#list_subcollections_of_a_document) specifically exclude the feature for "mobile/web client" libraries.  I wouldn't think Python was one of them, but we don't support the feature.  Do you know if that was intended?

**Comment by jba** [comment]
@schmidt-sebastian will know.

**Comment by schmidt-sebastian** [comment]
This is not intended and should be added to the Python SDK.

**Comment by tseaver** [comment]
@theacodes, @schmidt-sebastian, @jba Can any of you tell me which project we are using to CI firestore?  It can't be done using our normal, datastore-enabled projects.

**Comment by schmidt-sebastian** [comment]
You have to create a new project and enable Firestore without enabling Datastore. The easiest way to do this is through the Firebase Console.

**Comment by jba** [comment]
Presumably Python has such a project.

**Comment by tseaver** [comment]
Right, there is one being used for CI, but the credentials (including the project) are stored encrypted, and decrypted using CircleCI env vars.  I'm not sure who on the team knows the actual project ID. @dhermes, maybe?

**Comment by tseaver** [comment]
Projects to which I have access on https://console.cloud.google.com/:

```
google-cloud-node-no-whitelist | gcloud-node-whitelist-ci-tests
google-cloud-python | precise-truck-742
JJG Cloud Research | jjg-cloud-research
tresseaver-gcloud-research | citric-celerity-697
```

Of those, the main Python CI one (`precise-truck-742`) and @jgeewax's personal project (`jjg-cloud-research`) have the Firestore API enabled, but neither can actually use it, because they also have the Datastore API enabled.

**Comment by tseaver** [comment]
@crwilcox Maybe you have the keys?

**Comment by tseaver** [comment]
@jba Is the desired behavior just to list sub-collection IDs, or should the method return references to the subcollections?

**Comment by jba** [comment]
There's no conformance test, but there is a clear description of what should be returned in [the client API doc](https://docs.google.com/document/d/1PjoomZav75_wevZxTarHWqkAlcLnNBAhLBJ4NN_eCPs/edit#heading=h.gvhbp6ce4qht):
> The getCollections() method returns a list of results of CollectionReferences that are direct subscollections of this document.

**Comment by tseaver** [comment]
@jba OK, thanks -- I must've been misreading the [snippets on the "Getting Data" doc](https://cloud.google.com/firestore/docs/query-data/get-data#list_subcollections_of_a_document).

**Comment by guanzo** [comment]
Thank you!

**Comment by Rajczyk** [comment]
When will this be released on pypi?

**Comment by danoscarmike** [comment]
@tseaver I don't think there was a Firestore release cut since #5613 was merged. Could you prepare one please? Thanks!
